src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,359 @@
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { Button, Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import CurrentPostDetailsProvider from './provider';
+import { PostPerformanceData } from './post-performance-data';
+
+// Number of attempts to fetch the data before displaying an error.
+const FETCH_RETRIES = 3;
+
+/**
+ * Specifies the form of component props.
+ */
+interface PostDetailsSectionProps {
+	data: PostPerformanceData;
+}
+
+/**
+ * Outputs the current post's details or shows an error message on failure.
+ */
+function CurrentPostDetails() {
+	const [ loading, setLoading ] = useState<boolean>( true );
+	const [ error, setError ] = useState( null );
+	const [ postDetailsData, setPostDetails ] = useState<PostPerformanceData>( null );
+	const provider = new CurrentPostDetailsProvider();
+
+	useEffect( () => {
+		const fetchPosts = async ( retries: number ) => {
+			provider.getCurrentPostDetails()
+				.then( ( result ) => {
+					setPostDetails( result );
+					setLoading( false );
+				} )
+				.catch( async ( err ) => {
+					if ( retries > 0 ) {
+						await new Promise( ( r ) => setTimeout( r, 500 ) );
+						await fetchPosts( retries - 1 );
+					} else {
+						setError( err );
+						setLoading( false );
+					}
+				} );
+		};
+
+		setLoading( true );
+		fetchPosts( FETCH_RETRIES );
+	}, [] );
+
+	if ( error ) {
+		// "Soft" error for which we do not want to show an "Error:" prefix.
+		if ( 'string' === typeof error ) {
+			return <p>{ error }</p>;
+		}
+
+		// Error coming from apiFetch.
+		if ( error?.message ) {
```

Comment:
Type annotation for `useState` should allow confidence that we have an Error object
```suggestion
		if ( error.message ) {
```